src/compiler/sys.ts

```diff
@@ -100,6 +109,34 @@ module ts {
                 }
             }
 
+            function getNames(collection: any): string[] {
+                var result: string[] = [];
+                for (var e = new Enumerator(collection); !e.atEnd(); e.moveNext()) {
+                    result.push(e.item().Name);
+                }
+                return result.sort();
+            }
+
+            function readDirectory(path: string, extension?: string): string[] {
+                var result: string[] = [];
+                visitDirectory(path);
+                return result;
+                function visitDirectory(path: string) {

```

Comment:
Everything is a function within a function in our code base. What we need to look out for is nested functions that create closures and/or nested functions in outer functions that are invoked often. In this case neither applies.